### PR TITLE
[batch] Single db operation for inserting attempt resources

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

File: batch/batch/driver/main.py [hunk 1]
```diff
@@ -1324,6 +1337,17 @@ async def on_startup(app):
...
+    resource_ids = {
+        record['resource']: record['resource_id']
+        async for record in db.select_and_fetchall(
+            '''
+SELECT resource, resource_id FROM resources;
+'''
+        )
+    }
+
+    app['resource_name_to_id'] = resource_ids
```
Comment (daniel-goldstein) [inline batch/batch/driver/main.py:1349]: Nit: Could you call `refresh_globals_from_db` here instead of repeating the query?

**Review by daniel-goldstein — DISMISSED** [review]
So small and quick!

**Review by daniel-goldstein — APPROVED** [review]

**Comment by jigold** [comment]
A billing limit test failed which is concerning. I'm going to get rid of the approval until I understand why.

**Comment by jigold** [comment]
I'm pretty sure I fixed the issue. Yay for tests!

**Review by daniel-goldstein — APPROVED** [review]

**Pull request merged**